airflow/providers/google/cloud/hooks/compute_ssh.py

```diff
@@ -303,9 +387,29 @@ def _authorize_os_login(self, pubkey):
         self.log.info("Importing SSH public key using OSLogin: user=%s", username)
         expiration = int((time.time() + self.expire_time) * 1000000)
         ssh_public_key = {"key": pubkey, "expiration_time_usec": expiration}
-        response = self._oslogin_hook.import_ssh_public_key(
-            user=username, ssh_public_key=ssh_public_key, project_id=self.project_id
-        )
+        if self._check_running_in_parallel():
+            self.log.info("Several processes are executed in parallel")
+            bucket_name = f"tmp_bucket_{self.project_id}"
+            lock_file_name = "file.lock"
+            lock_path = f"gs://{bucket_name}/{lock_file_name}"
+            self._create_bucket_for_lock(bucket_name=bucket_name)
+
+            max_time_to_wait = 10
+            for time_to_wait in exponential_sleep_generator(initial=1, maximum=max_time_to_wait):
+                if self._acquire_lock(lock_path=lock_path):
+                    try:
+                        response = self._oslogin_hook.import_ssh_public_key(
+                            user=username, ssh_public_key=ssh_public_key, project_id=self.project_id
+                        )
+                    finally:
+                        self._release_lock(lock_path=lock_path)
+                        break
+                self.log.info("Another process is running, waiting...")
+                time.sleep(time_to_wait)
```

Comment:
What happens if this for-loop is over and we never acquired the lock? Should we raise an exception or log warning at least, because in this case the execution continues without calling the `self._oslogin_hook.import_ssh_public_key(...)`?